activesupport/lib/active_support/encrypted_configuration.rb

```diff
@@ -39,7 +39,13 @@ def options
       end
 
       def deserialize(config)
-        config.present? ? YAML.load(config, content_path) : {}
+        return {} if config.blank?
+
+        if Gem::Version.new(Psych::VERSION) >= Gem::Version.new("3.0.1")
+          YAML.load(config, fallback: YAML.parse("{}"))
```

Comment:
Can we pass a `{}` in the fallback or does it need to parse YAML?

Reply:
Unfortunately just passing `{}` is not possible. It is not described in the documentation but the fallback needs to be an object that responds to `to_ruby` - like an instance of `Psych::Nodes::Document` that is returned by `YAML.parse`.

Reply:
How about using `presence`? For example as follows. 

```ruby  
def deserialize(config)
  YAML.load(config).presence || {}
end
```

Reply:
@y-yagi Unfortunately `YAML.load(config).presence || {}` is not possible because `YAML.load(config)` return `false` for empty or comment-only YAML strings - and `false.presence || {}` evaluates to `false`.

Reply:
`false.presence` is nil, so `false.presence || {}` evaluates to `{}`.

Reply:
Ups, I am sorry, you are right. `false.presence` returns of cause `nil`.

`YAML.load(config).presence || {}` might be an option if the goal is to return an empty hash if `YAML.load` returns something that is not useful in this context in which it would raise a `NoMethodError: undefined method `deep_symbolize_keys'`.

I used `YAML.load(config, fallback: YAML.parse("{}"))` instead because it returns the fallback only if the YAML was blank or comment-only. If there is any YAML content it would return the content of the file - even it is was `false` or `nil`.

I wonder what is the least surprising response? 

To return always the `{}` default when `YAML.parse` returns blank - as George supposes? Or a `{}` default only for empty files, but an error for well-formed but blank values - as my current version?

What Do you think is better? I am open to both.

Reply:
This is getting too theoretical. The case we know about is parsing the default all-comment template, so let's fix that the simplest way we can. Then defer future fixes to if we encounter those (currently unknown) cases.

The `:fallback` option branching on the version is too complex a fix for this. Let's go with `presence`.

Reply:
@kaspth Thanks for making clear what to focus on.